sklearn/utils/validation.py

```diff
@@ -407,19 +407,30 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         dimensions or is originally 1D and ``ensure_2d`` is True. Setting to 0
         disables this check.
 
-    warn_on_dtype : boolean (default=False)
+    warn_on_dtype : boolean or None, optional (default=None)
         Raise DataConversionWarning if the dtype of the input data structure
         does not match the requested dtype, causing a memory copy.
 
+        ..deprecated:: 0.21
+            "``warn_on_dtype`` is deprecated in "
+            "version 0.21 and will be removed in 0.23."
+
     estimator : str or estimator instance (default=None)
         If passed, include the name of the estimator in warning messages.
 
     Returns
     -------
     array_converted : object
         The converted and validated array.
-
     """
+    # warn_on_dtype deprecation
+    if warn_on_dtype is not None:
+        warnings.warn(
+            "'warn_on_dtype' is deprecated in "
+            "version 0.21 and will be removed in 0.23."
```

Comment:
```suggestion
            "version 0.21 and will be removed in 0.23. "
```

Needs a space after period (same for the other)

Reply:
done